src/handler.jl

```diff
@@ -23,22 +23,19 @@ Names with a '.' separators will be used to form subdirectories
 (e.g., "Foo.bar.x" will be saved to "\$prefix/Foo/bar/x.jlso").
 """
 function path(handler::Handler{P}, name::String; tags...) where P
-    # deal with dynamic scope tags and checkpoint tags
-    joint_tags = intersect(keys(TAGS[]), keys(tags))
-    isempty(joint_tags) || throw(ArgumentError("Application- and package-level tags both contain $(joint_tags)."))
-    all_tags = collect(Iterators.flatten((TAGS[], tags)))
+    with_tags(tags...) do
+        # Build up a path prefix based on the tags passed in.
+        prefix = Vector{String}(undef, length(CONTEXT_TAGS[]))
+        for (i, t) in enumerate(CONTEXT_TAGS[])
+            prefix[i] = string(first(t), "=", last(t))
+        end
```

Comment:
I know you didn't write this code, but it is pretty weird right now.
I guess it was written before folks at invenia knew about list comprehensions, or `map`.
```suggestion
        prefix = ["$key=$val" for (key,val) in CONTEXT_TAGS]
```